runtime/src/qio/qio_popen.c

```diff
@@ -337,9 +338,10 @@ qioerr qio_waitpid(int64_t pid,
 
   if( WIFEXITED(status) ) {
     *exitcode = WEXITSTATUS(status);
-    if( WIFSIGNALED(status) ) {
-      *exitcode = - WSTOPSIG(status);
-    }
+    *done = 1;
+  }
+  else if( WIFSIGNALED(status) ) {
+    *exitcode = WTERMSIG(status);
```

Comment:
I missed your intent with the negative exit code for indicating abnormal termination by signal. I agree that's a good feature.